### PR TITLE
Public booking ui

### PR DESCRIPTION
**Pull request by stevepetebruce**
#### What's this PR do?
Added UI for public guest booking and stripe.js for taking secure card payments.
Configured Babel in webpacker to compile Async/Await.

#### Where should the reviewer start?
/public/trips/trip_id/bookings/new

#### Screenshots
screenshot 2018-11-04 at 13 37 34

---

### CODE REVIEW

File: app/javascript/javascript/custom.js [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
 // Keep all Stripe related JS in one file, and only load that file on sensitive pages.
-// Eventually we can move most/all JS into Stimulus JS, ref: https://stimulusjs.org/
\ No newline at end of file
+// Eventually we can move most/all JS into Stimulus JS, ref: https://stimulusjs.org/
+
+(function() {
+  "use strict";
+
+  document.addEventListener("DOMContentLoaded", function(){
+    const stripe = Stripe("pk_test_mOBtxNpMht2po09RkLReWILl");
```
Comment (AlanDonohoe) [inline app/javascript/javascript/custom.js]: Will need to keep all keys out of Github, using dot.env

Comment (AlanDonohoe) [reply app/javascript/javascript/custom.js]: https://www.newscientist.com/article/2168713-huge-new-facebook-data-leak-exposed-intimate-details-of-3m-users/

Comment (AlanDonohoe) [reply app/javascript/javascript/custom.js]: https://github.com/bkeepers/dotenv

Comment (stevepetebruce) [reply app/javascript/javascript/custom.js]: Hi Al, I made a note of that in the commit. This is the test key so will need the dot.env file when we convert to the production key.

Comment (stevepetebruce) [reply app/javascript/javascript/custom.js]: There are two warning issues remaining. These are from official documentation and break if altered. Are we able to force these codacy issues through?

Comment (AlanDonohoe) [reply app/javascript/javascript/custom.js]: Yeah I think we can either set up Codacy to ignore certain rules or we can merge this in as an Admin - like a forced merge.

Let me move the stripe key out and recommit.

Our local and staging environments will have the test stripe key and production will have the live key. Dot.env allows that pattern, which you couldn't do with it hard coded in.

File: app/views/public/trips/bookings/_form.html.erb [hunk 2]
```diff
@@ -1,21 +1,43 @@
-<%= form_with(model: booking, local: true, url: url) do |form| %>
+<%= form_with(model: booking, local: true, url: url, :html => {:id => "payment-form"}) do |form| %>
```
Comment (AlanDonohoe) [inline app/views/public/trips/bookings/_form.html.erb]: Ruby / Rails stuff but still... 

Can use new format for symbols in key value pairs:

Was: 
:html => {:id => "payment-form"}
Now:
html: { id: "payment-form" }
// Note also, the coding standard of leaving a space in between the start and end of a hash
ref: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/SpaceInsideHashLiteralBraces

File: app/javascript/javascript/custom.js (hunk 1 above)
Comment (AlanDonohoe) [reply app/javascript/javascript/custom.js]: Have added dot env now -  all good.

**Comment by AlanDonohoe** [comment]
Don't know where the codacy results are for this one? Seems like the only PR that doesn't have Codacy?

File: app/views/public/trips/bookings/_form.html.erb (hunk 2 above)
Comment (stevepetebruce) [reply app/views/public/trips/bookings/_form.html.erb]: Updated Rails coding standard.

**Pull request merged**